### PR TITLE
Verify schema name length when creating or renaming in Postgres

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Verify schema name length when creating or renaming in Postgres

## Documentation

(x) No documentation is needed.

## Release notes

(x) Release notes entries required with the following suggested text:

```markdown
# PostgreSQL
* Prevent creating or renaming to a schema with a name longer than the max length.
  Previously, the name was truncated to the max length. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseConnectorTest.java [hunk 1]
```diff
@@ -693,6 +694,43 @@ public void testCreateTableWithLongTableName()
         assertTrue(getQueryRunner().tableExists(getSession(), validTableName));
     }
 
+    @Override
+    public void testRenameSchemaToLongName()
+    {
+        // Override because the max length is different from CREATE TABLE case
```
Comment (hashhar) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseConnectorTest.java:700]: The comment feels stale. Or did you mean `different from CREATE SCHEMA case`?

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java [hunk 2]
```diff
@@ -518,6 +518,18 @@ protected String createSchemaSql(String schemaName)
...
+    @Override
+    protected OptionalInt maxSchemaNameLength()
+    {
+        return OptionalInt.of(128);
+    }
+
+    @Override
+    protected void verifySchemaNameLengthFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessageMatching("(?s)(.*Read timed out)|(.*\"`NAME`\" that has maximum length of 128.*)");
```
Comment (hashhar) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java:530]: Is `Read timed out` a flaky issue or related to larger schema names?

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseConnectorTest.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseConnectorTest.java:700]: Yes, let me fix the comment.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java (hunk 2 above)
Comment (ebyhr) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java:530]: It relates to larger schema names. (Using long name increases the possibility of read timed out)

**Review by hashhar — APPROVED** [review]

**Comment by ebyhr** [comment]
Rebased on upstream and remove `BaseHiveConnectorTest.testRenameSchemaToLongName` to resolve logical conflicts.

**Pull request merged**